### PR TITLE
router: match longest entrypoint path prefix

### LINKED ISSUE #388: Router should use longest valid pathPrefix match for sandbox entrypoints
The router currently picks the first entrypoint where `strings.HasPrefix(path, ep.Path)` is true.

That breaks overlapping prefixes. If a runtime has both `/` and `/api`, then `/api/foo` can get routed to `/` depending on the order in the CR. Also `/api2` matches `/api`, which is not really the documented `/api/...` behavior.

Steps:
1. Create an AgentRuntime/CodeInterpreter with two ports:
   - `/`
   - `/api`
2. Put `/` before `/api` in the spec.
3. Send a request to `/api/foo`.

Expected:
Request goes to the `/api` entrypoint.

Actual:
Request can go to the `/` entrypoint.

This should probably do longest-prefix matching and make sure `/api` only matches `/api` or `/api/...`, not `/api2`.

### PR DESCRIPTION
**Pull request by avinxshKD**
/kind bug

**What this PR does / why we need it**:

Fixes router entrypoint selection when multiple `pathPrefix` values overlap.

The router was picking the first prefix match, so a runtime with `/` before `/api` could route `/api/foo` to the `/` entrypoint. It also treated `/api2` as matching `/api`.

This changes the selection to use the longest valid prefix match and keeps path boundaries sane.

**Which issue(s) this PR fixes**:
Fixes #388

**Special notes for your reviewer**:

Added focused tests for overlapping prefixes and `/api2` boundary behavior.

**Does this PR introduce a user-facing change?**:

```release-note
Router now selects the longest valid sandbox entrypoint path prefix when routing invocation requests.

### CODE REVIEW

**Comment by avinxshKD** [comment]
Hey @LiZhenCheng9527 @hzxuzhonghu checked this while looking at the router path selection logic.

Current code picks the first HasPrefix match, so /can steal  /api/foo depending on spec order. Also /api2 matches /api.
Longest valid prefix match should fix this cleanly.

Pls take a look when free.

File: pkg/router/handlers.go [hunk 1]
```diff
@@ -120,6 +125,13 @@ func determineUpstreamURL(sandbox *types.SandboxInfo, path string) (*url.URL, er
 	return buildURL(ep.Protocol, ep.Endpoint)
 }
 
+func entryPointPathMatches(path, prefix string) bool {
+	if prefix == "" || prefix == "/" {
+		return true
+	}
+	return path == prefix || strings.HasPrefix(path, prefix+"/")
+}
```
[the remark this answers is not in the document]
Comment (avinxshKD) [reply pkg/router/handlers.go:138]: Good catch. Updated this to normalize the incoming path and trim trailing slashes from the configured prefix.

File: pkg/router/handlers.go [hunk 2]
```diff
@@ -106,12 +106,17 @@ func (s *Server) handleGetSandboxError(c *gin.Context, err error) {
 }
 
 func determineUpstreamURL(sandbox *types.SandboxInfo, path string) (*url.URL, error) {
-	// prefer matched entrypoint by path
+	var matched types.SandboxEntryPoint
+	found := false
```
[the remark this answers is not in the document]
Comment (avinxshKD) [reply pkg/router/handlers.go:111]: Leaving this out for now. `determineUpstreamURL` is only called after session lookup returns a sandbox, so a nil guard here feels unrelated to the path matching fix.

File: pkg/router/handlers_test.go [hunk 3]
```diff
@@ -241,6 +241,55 @@ func TestHandleInvoke_NoEntryPoints(t *testing.T) {
...
+		{
+			name: "longest prefix wins",
+			path: "/api/run",
+			entryPoints: []types.SandboxEntryPoint{
+				{Endpoint: "10.0.0.1:8080", Protocol: "HTTP", Path: "/"},
+				{Endpoint: "10.0.0.2:8080", Protocol: "HTTP", Path: "/api"},
+			},
+			wantHost: "10.0.0.2:8080",
+		},
+		{
```
Comment (acsoto) [inline pkg/router/handlers_test.go:260]: This test only covers the boundary case when a `/` entrypoint exists. With only `/api` configured, `/api2/run` still falls through to the first entrypoint, so it can still route to `/api`; is that fallback behavior intended?

Comment (avinxshKD) [reply pkg/router/handlers_test.go:260]: i agree, added a case for that.

With only `/api` configured, `/api2/run` does not match `/api`, it falls back to the first entrypoint, same as the existing router behavior. This change only affects which matched prefix wins. PTAL

**Comment by avinxshKD** [comment]
@RainbowMango ptal

**Comment by avinxshKD** [comment]
@RainbowMango PTAL. its been marked as lgtm. can we merge this??
